addons/forum-post-countdown/userscript.js

```diff
@@ -0,0 +1,37 @@
+export default async function ({ addon, msg }) {
+  const submitButton = document.querySelector("#djangobbwrap .form-submit [type=submit]");
+
+  submitButton.addEventListener("click", () => {
+    if (!localStorage.getItem("sa-forum-post-countdown")) {
+      localStorage.setItem("sa-forum-post-countdown", Date.now());
+    }
+  });
+
+  const countdown = localStorage.getItem("sa-forum-post-countdown");
+  if (!countdown) {
+    return;
+  }
+
+  const loggedIn = await addon.auth.fetchIsLoggedIn();
+  const secondCount = (
+    await fetch("/session", { headers: { "x-requested-with": "XMLHttpRequest" } }).then((resp) => resp.json())
+  ).permissions.new_scratcher
+    ? 120
+    : 60;
+
+  const elt = document.createElement("div");
+  elt.id = "sa-forum-post-countdown";
+  document.body.appendChild(elt);
+
+  setInterval(async () => {
+    const now = Date.now();
+    if (now > Number(countdown) + secondCount * 1000 || !loggedIn) {
```

Comment:
`loggedIn` isn't going to change with time, so it doesn't make sense to use its value on the interval.

Reply:
Good catch